### PR TITLE
Ignore unrecognized arguments

### PR DESCRIPTION
**Pull request by gaborbarna**

### CODE REVIEW

**Comment by gaborbarna** [comment]
@alexarchambault Hey, what do you think about this change?

File: cats/shared/src/main/scala/caseapp/cats/IOCaseApp.scala [hunk 1]
```diff
@@ -65,11 +65,20 @@ abstract class IOCaseApp[T](implicit val parser0: Parser[T], val messages: Help[
   def stopAtFirstUnrecognized: Boolean =
     false
 
+  /**
+    * Whether to ignore unrecognized arguments.
+    *
+    * That is, if there are unrecognized arguments, the parsing still succeeds.
+    * The unparsed arguments are put in the `args` argument of `run`.
+    */
+  def ignoreUnrecognized: Boolean =
```
Comment (alexarchambault) [inline cats/shared/src/main/scala/caseapp/cats/IOCaseApp.scala:74]: I believe this parameter should be taken into account in `def parser` above, and `Parser.withHelp` too. The call to `parser.withHelp` below returns a `Parser` wrapping `parser`, and the `ignoreUnrecognized` parameter can only gets through thanks to this, IIRC.

The tests of case-app are missing tests for `CaseApp` and `IOCaseApp` themselves. Passing those parameters this way is required to have them work.

**Review by alexarchambault — COMMENTED** [review]
Thanks! I only left a comment, about under-tested aspects of case-app.

File: cats/shared/src/main/scala/caseapp/cats/IOCaseApp.scala (hunk 1 above)
Comment (gaborbarna) [reply cats/shared/src/main/scala/caseapp/cats/IOCaseApp.scala:74]: Thanks for the quick feedback, I'll address this soon.

File: tests/shared/src/test/scala/caseapp/demo/Demo.scala [hunk 2]
```diff
@@ -123,18 +123,27 @@ object ManualCommandNotAdtStuff {
     }
   }
 
+  case object Command5IgnoreUnrecognized extends CaseApp[ManualCommandNotAdtOptions.Command5Opts] {
```
Comment (alexarchambault) [inline tests/shared/src/test/scala/caseapp/demo/Demo.scala:126]: I had forgotten about that file ^^ That's where this should have been tested 👍

**Review by alexarchambault — APPROVED** [review]
Thanks!

**Comment by gaborbarna** [comment]
@alexarchambault Thanks, can you please release the latest version to maven?

**Comment by alexarchambault** [comment]
I had cut a release, but it seems something went wrong during the [release job](https://travis-ci.org/github/alexarchambault/case-app/jobs/765058348). I'll have a look later this week I think.

**Comment by gaborbarna** [comment]
> I had cut a release, but it seems something went wrong during the [release job](https://travis-ci.org/github/alexarchambault/case-app/jobs/765058348). I'll have a look later this week I think.

Hey, any news on that front?

**Pull request merged**